vm/runtime.ts

```diff
@@ -1271,6 +1279,10 @@ module J2ME {
     return address;
   }
 
+  export function freeMemory(): number {
```

Comment:
Nit: *free* is both an adjective and a verb, and function names often start with verbs, although *free* is actually being used as an adjective here, which makes this name confusing; so I would call it *getFreeMemory*.